app/Http/Controllers/ContactController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\DB;
+
+use App\Contact;
+
+class ContactController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $contacts = DB::table('contacts')->get();
+
+        foreach ($contacts as $contact) {
+            try {
+                $response = file_get_contents(env('EMAIL_SUBSCRIPTION_API') . '/get_status?email=' . $contact->email);
+                $data = json_decode($response);
+                if ($data['status'] == 'subscribed') {
+                    $contact->subscribed = true;
+                } else {
+                    $contact->subscribed = false;
+                }
+            } catch (\Exception $e) {
+                Log::error($e);
```

Comment:
[Good] This is real cool. Logging error helps to really understand what has happen to the application at some certain times and help in debugging